### DISCUSSION TITLE
Get mime type from uploaded video

### DISCUSSION THREAD

**Discussion by wpas81** [filamentphp/filament#5514] [Help] [answered]
Hi, 

I'm taking my first steps in Filament & having some trouble figuring out a few tricks. 
eg I'm having a form with a video-upload and I want to store the mime-type in a separate column. How do I set the mime-type-field to the mime of the uploaded video?

I assumed I could do it via $get('video')->getMimeType() but that throws the error "Call to a member function getMimeType() on null"

```
return $form
    ->schema([
        Forms\Components\TextInput::make('title')
            ->required()
            ->maxLength(255)
            ->autofocus(),
        
        FileUpload::make('path')
            ->label('Video')
            ->acceptedFileTypes(['video/mp4','video/ogg','video/webm'])
            ->maxSize(51200)
            ->disk('videos')
            ->visibility('private')
            ->afterStateUpdated(fn (callable $set, callable $get) => $set('mime', $get('video')->getMimeType())),
        
        Forms\Components\TextInput::make('mime')
            ->maxLength(25),               
        ]);
```

**Comment by danharrin** [comment, marked as answer]
Try this maybe:

`->afterStateUpdated(fn (callable $set, $state) => $set('mime', $state?->getMimeType())),`

**Reply by wpas81** [reply]
thanks! I just found it myself too :)

**Reply by MansourM** [reply]
thank you for the solution,
but as i understand it this needs us to have a Hidden `mime` field or it won't go through validation,
but that's not my issue i was hoping to be able to check this server-side since those tend to be much safer, instead of setting a hidden field and sending the data through client.

I am new to filament, so i am probably misunderstanding something, but i searched through documents a couple of times and could not find a better solution that the one you mentioned.
